Assets/Scripts/CreatureStats.cs

```diff
@@ -0,0 +1,76 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public abstract class CreatureStats : MonoBehaviour
+{
+    public Dictionary<StatType, float> _stats;
+
+    void Start()
+    {
+        // Choose stats for creature
+        // _stats = new Dictionary<StatType, float>()
+        // {
+        //     { StatType.Health, 100f },
+        //     { StatType.MaxHealth, 100f },
+        //     { StatType.ArmorPercentage, 0f },
+        // };
+    }
+
+    public float ValidateStat(StatType statType, float statValue)
```

Comment:
We can remove this by making Dictionary<StatType, StatHandler> where the last one has set method/property and will validate everything by itself. Also it may help implement types other than float